### ISSUE TITLE
Fix Windows build clients for Anaconda Cloud

### ISSUE THREAD

**Issue by ijstokes** [Anaconda-Server/anaconda-build#150]
Windows build workers for the Anaconda Build System on Anaconda Cloud are not currently working.  They need to be fixed so we can do automated cross platform package builds.

**Comment by ijstokes** [comment]
I'm told @tpowellcio is looking into this.  If possible, it would be super helpful if Anaconda-Server/anaconda-notebook#56 could be updated when this is resolved and closed.

**Comment by mmarchetti** [comment]
win-64 python 2.7 builds fail this way:
112IOError: [Errno 13] Permission denied: u'C:\\Users\\binstar\\Miniconda\\pkgs\\python-2.7.11-0\\DLLs\\_ctypes.pyd'
113Internal binstar build error: Could not set up initial build state

https://anaconda.org/anaconda-notebook/nb_locker/builds/7/6

**Comment by mmarchetti** [comment]
this is 100% recreatable, all windows python=2.7 builds seem to be failing. May be related to python 2.7.11?

**Comment by mmarchetti** [comment]
Is there any outlook on a fix for this issue? We're trying to wrap up work on our Jupyter extensions epic this week and automated build is one of the last pieces we need. We're blocked on this issue.

@Anaconda-Server/anaconda-repository

**Comment by stephenakearns** [comment]
@trentoliphant made an update today - @mmarchetti testing...

**Comment by PeterDSteinberg** [comment]
I'll look into this now.

**Comment by stephenakearns** [comment]
Hi @PeterDSteinberg  - re-assigned this item to you based on your comments above. LMK if that doesn't work for you.

**Comment by PeterDSteinberg** [comment]
I apologize.  I have dropped the ball on this issue that was assigned to me.  I worked to the point of getting this error replicated on anaconda-build with Python 2.7 on a Windows VirtualBox VM, but I haven't found a fix.  I filed the conda issue mentioned just above (1936).  The comments on that issue mentioned it was a zombie process problem and @srossross may have worked on a solution?

**Comment by damianavila** [comment]
@PeterDSteinberg @srossross @stephenakearns any update on this issue? Thanks!

**Comment by PeterDSteinberg** [comment]
@damianavila I worked on this Friday and didn't resolve.  It's on my immediate TODO list for this afternoon / tomorrow.

**Comment by damianavila** [comment]
Thanks for the update @PeterDSteinberg.

**Comment by stephenakearns** [comment]
Hi @PeterDSteinberg  and @srossross - I think you indicated earlier that this issue was resolved. Can you update this issue?

**Comment by stephenakearns** [comment]
Hi @PeterDSteinberg - can you provide latest status update on this and attach and recent PRs associated with it. Thx

**Comment by bollwyvl** [comment]
Here's an another issue with this: https://github.com/Anaconda-Server/nb_conda/pull/9#issuecomment-179265547

**Comment by stephenakearns** [comment]
@PeterDSteinberg added some additional diagnostics info to Conda - conda/conda#2023

**Comment by stephenakearns** [comment]
Example of working Win-64 Build: https://anaconda.org/binstar/binstar/builds/497

**Comment by damianavila** [comment]
Great, I will test with our extensions... thanks @stephenakearns

**Comment by damianavila** [comment]
@stephenakearns not fixed for us... still seeing the same issue, check this build just triggered morning today: https://anaconda.org/anaconda-nb-extensions/nb_conda/builds/18/9

**Comment by damianavila** [comment]
well I am not sure if this wes even deployed because I see the PR still opened, am I right?

**Comment by PeterDSteinberg** [comment]
@srossross @stephenakearns @insertinterestingnamehere I have had some successful builds with a testing deployment of win-64.

Here is what I did.  I started an EC2 win-64 instance from the `binstar-build-win-64-1.0.0-ec2(ami-b2f4a0da)` image.  That machine already had git, visual studio tools, and 2 miniconda installs: binstar user's python at `C:\Users\binstar\Miniconda` and Adminstrator's python at `C:\Miniconda`.

Installed `chalmers`, `anaconda-client` develop branches as Admin into the C:\Miniconda environment.  Installed also [anaconda-build PR 197](https://github.com/Anaconda-Server/anaconda-build/pull/197) into the Admin python.

Ran the following commands as the binstar user, then *again* as the Admin user:
```
conda config --set always_yes true
conda update conda
conda install pip
conda install -c https://conda.anaconda.org/tpowell unxutils
conda install conda-build
conda install pywin32 psutil mock patch
conda install python=2.7.10
```

We should see at this point that when logged in as binstar user, the C:\Users\binstar\Miniconda is on the PATH and when logged in as admin C:\Miniconda is on the PATH.

As the Admin user in a cmd session, set some env variables useful for worker data:
```
set HOME_DIR=C:\Users\binstar
set WORKER_ID="win-64-psteinberg-aws-4"
set TOKEN_FILE=%HOME_DIR%\.anaconda.token
set QUEUE=psteinberg/abc
```
Then do these commands, which may require pasting one by one to avoid user input errors on passwords.  These register a worker and only need to be done once in the event of worker restarts:
```
anaconda auth --create -n "anaconda-build-win-64-psteinberg-99o" --scopes "api:build-worker api:read api:build" --out %TOKEN_FILE%
python -c "import socket;print socket.gethostname()" > %HOME_DIR%\HOSTNAME.txt

set /p HOSTNAME=<%HOME_DIR%\HOSTNAME.txt
anaconda login
anaconda worker register -p win-64 --dist aws-win64 --hostname %HOSTNAME%  %QUEUE% --name %WORKER_ID%
anaconda logout
```

Now, as binstar user in a session, redefine the env variables:
```
```
Then use those env vars to use the correct python and chalmers to run the builds.  Note that as the binstar user, `chalmers` as a CLI item is not defined, and that's probably good.
```
set APY=C:\Miniconda\python.exe
set AA=C:\Miniconda\Scripts\anaconda-script.py
set AC=C:\Miniconda\Scripts\chalmers-script.py
%APY% %AC% stop --all --force
%APY% %AC%  remove --all
%APY% %AC%  add --name worker -c "%APY% %AA% --show -t %TOKEN_FILE% worker run %WORKER_ID% --show-new-procs --status-file %HOME_DIR%\worker\worker.status"
%APY% %AC%  start --all
```

### Follow up comments
 * Note I had to downgrade python on both binstar and Admin to 2.7.10
 * The build worker still encounters broken build initialization if a build explicitly requests a Python 2.7.10 install (files lock issue conflicting with the build worker on write of Python 2.7.10 DLLs) or if the binstar and Admin Python are upgraded to 2.7.11, the default minor version of most Python 2.7 builds (files lock issue on Python 2.7.11 DLLs).
 * The PR #189 if installed into this 2-Python setup and run with the commands above, results in failure to start the worker for hitting the runtime check (There were two other `C:\Miniconda\python.exe` processes running at startup [with no further info]), unless the `--ignore-process-check` flag is not given.

**Comment by stephenakearns** [comment]
Hi @PeterDSteinberg - when the other build worker issues are resolved, can you connect with either @kwmsmith (Blaze/Dataportal) or @damianavila (Enterprise Notebooks) to validate the latest fix to the Windows Workers

**Comment by PeterDSteinberg** [comment]
@kwmsmith and @damianavila 

From the information in this and related issues we have:
 * modified windows deployment in the wiki based on issue #198 
 * found that most of the issue was related to conda installation's tarball extract command causing changes to running executables

**Comment by PeterDSteinberg** [comment]
Closing this issue as it is now tracked by [conda #2079](https://github.com/conda/conda/issues/2079)
